test/extended/apiserver/resiliency.go

```diff
@@ -0,0 +1,65 @@
+package apiserver
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/uuid"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	"github.com/openshift/origin/test/extended/util/disruption"
+	"github.com/openshift/origin/test/extended/util/disruption/controlplane"
+	types "k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+func forceApiRollout(oc *exutil.CLI) {
+	redeploymentReason := fmt.Sprintf(`{"spec":{"forceRedeploymentReason":"resiliency-test-%v"}}`, uuid.NewUUID())
+
+	_, err := oc.AdminOperatorClient().OperatorV1().KubeAPIServers().Patch(context.Background(), "cluster", types.MergePatchType,
+		[]byte(redeploymentReason), metav1.PatchOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+}
+
+func isApiUp(oc *exutil.CLI) (bool, error) {
+	_, err := oc.AdminKubeClient().CoreV1().Namespaces().List(context.TODO(), metav1.ListOptions{})
+	return err == nil, err
+}
+
+var _ = ginkgo.Describe("[Conformance][sig-api-machinery][Serial] Cluster resiliency", func() {
+	f := framework.NewDefaultFramework("single-node")
+	f.SkipNamespaceCreation = true
+	f.SkipPrivilegedPSPBinding = true
+
+	oc := exutil.NewCLIWithoutNamespace("cluster-resiliency")
+
+	ginkgo.It("should allow a fast kube-apiserver rollout", func() {
+		disruption.Run(f, "Cluster Resiliency to API shutdown", "cluster-resiliency", disruption.TestData{},
+			[]upgrades.Test{
+				&controlplane.ApiFastRestorationTest{},
+			},
+			func() {
+				ginkgo.By("Forcing API rollout")
+				forceApiRollout(oc)
+
+				ginkgo.By("Expecting API downtime")
+				wait.PollImmediate(3*time.Second, time.Minute, func() (bool, error) {
```

Comment:
What's the difference between this polling and the polling in ApiFastRestorationTest?

Reply:
so this part should just do he rollout and verify rollout is occurring
the ``ApiFastRestorationTest`` (and pod-watches which I'll add in the future) should include everything we want to actually measure

Reply:
ok, I removed it as an invariant
we'll first have the basic test and the pods-watching logic will join afterwards as an invariant